tests/Importers/MgFotoImporterTest.php

```diff
@@ -0,0 +1,331 @@
+<?php
+
+namespace Tests\Importers;
+
+use App\Import;
+use App\Importers\MgFotoImporter;
+use App\ImportRecord;
+use App\Item;
+use App\Matchers\AuthorityMatcher;
+use App\Repositories\CsvRepository;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Translation\Translator;
+use PHPUnit\Framework\MockObject\MockObject;
+use Tests\TestCase;
+use Tests\WithoutSearchIndexing;
+
+class MgFotoImporterTest extends TestCase
+{
+    use RefreshDatabase;
+    use WithoutSearchIndexing;
+
+    protected CsvRepository|MockObject $repositoryMock;
+
+    protected MgFotoImporter $importer;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->repositoryMock = $this->createMock(CsvRepository::class);
+        $this->importer = new MgFotoImporter(
+            app(AuthorityMatcher::class),
+            $this->repositoryMock,
+            app(Translator::class)
+        );
+    }
+
+    /** @dataProvider mediumProvider */
+    public function testMedium($material, $povrch, $translations)
+    {
+        $this->importData([
+            'Material' => $material,
+            'Povrch' => $povrch,
+        ]);
+
+        $item = Item::find('CZE:MG.rada_s_123-lomeni_s');
+        $this->assertEquals($translations['cs'], $item->translate('cs')->medium);
+        $this->assertEquals($translations['sk'], $item->translate('sk')->medium);
+        $this->assertEquals($translations['en'], $item->translate('en')->medium);
+    }
+
+    public static function mediumProvider()
+    {
+        return [
+            'material' => [
+                'papír',
+                null,
+                [
+                    'cs' => 'papír',
+                    'sk' => 'papier',
+                    'en' => 'paper',
+                ],
+            ],
+            'materialWithSurface' => [
+                'papír',
+                'M',
+                [
+                    'cs' => 'papír/matný',
+                    'sk' => 'papier/matný',
+                    'en' => 'paper/matte',
+                ],
+            ],
+        ];
+    }
```

Comment:
I've _finally_ gotten what the dataproviders do :D 

For what it's worth, I find this more legible and straightforward. Not much value in DRYing this up IMHO.

```suggestion
    public function testMedium()
    {
        $this->importData([
            'Material' =>'papír',
            'Povrch' => null,
        ]);

        $item = Item::find('CZE:MG.rada_s_123-lomeni_s');
        $this->assertEquals('papír', $item->translate('cs')->medium);
        $this->assertEquals('papier', $item->translate('sk')->medium);
        $this->assertEquals('paper', $item->translate('en')->medium);
        
        $this->importData([
            'Material' =>'papír',
            'Povrch' => 'M',
        ]);

        $item = Item::find('CZE:MG.rada_s_123-lomeni_s');
        $this->assertEquals('papír/matný', $item->translate('cs')->medium);
        $this->assertEquals('papier/matný', $item->translate('sk')->medium);
        $this->assertEquals('paper/matte', $item->translate('en')->medium);
    }
```

Reply:
The dataProvider here helps to keep the tests isolated though.